### DISCUSSION TITLE
Custom JavaScript does not work in _Layout file

### DISCUSSION THREAD

**Discussion by Nanouz** [Dotnet-Boxed/Templates#162] [Q&A]
Hi,

I am having problem running custom javascript in my _Layout file.
I made a Stackoverflow post but can't find a solution for it. (Click [here](http://stackoverflow.com/questions/43051288/asp-mvc-layout-page-does-not-run-custom-javascript))

Is this related to the template? I never had this before with any new project. Its the first time I use this.

Thx for any help!

**Comment by RehanSaeed** [comment]
You need to read about Content Security Policy (CSP). Inline scripts are blocked by default as a security feature.

**Comment by Nanouz** [comment]
I did not know that. Thx! The reason I use inline scripts is because I can use Razor in them. Is it a good practice to use hidden fields for certain values/properties I can't access in my JS file?

**Comment by RehanSaeed** [comment]
If you need to inject some values into your JavaScript, consider creating an endpoint that returns some JSON, call that using JavaScript and do what you need.

**Comment by Nanouz** [comment]
Hmmm I see. And what about if I want to do ajax calls from my view. How should I get the url in my JS file? I'm used to use '@Url.Action("","")' but that does not work in this case.

**Comment by RehanSaeed** [comment]
You can hard code these in JS or stick them in the JSON I mentioned above and use `@Url.Action("","")` to generate the JSON.

**Comment by Nanouz** [comment]
Do you have any example for the method you are suggesting? Thx for the help. Much appreciated.

**Comment by RehanSaeed** [comment]
https://stackoverflow.com/questions/42360139/asp-net-core-return-json-with-status-code
